discord/appinfo.py

```diff
@@ -68,6 +75,11 @@ def __init__(self, state, data):
         self.bot_require_code_grant = data['bot_require_code_grant']
         self.owner = User(state=self._state, data=data['owner'])
 
+        team = data.get('team')
+        self.team = Team(state, team) if team else None
+
+        self.game_info = GameInfo(state, data) if data['summary'] else None
```

Comment:
I don't really get the rationale for splitting this into two separate classes. I suppose it saves some extraneous key checks but this class name seems peculiar because it's not an actual API object. Not to mention that this key check seems suspicious and I feel like it tosses information aside as well.

Looking at the API documentation (https://discordapp.com/developers/docs/topics/oauth2#get-current-application-information) the following keys are always present regardless of the key being there or not:

- `summary`
- `verify_key`

I don't really want to lose information on the `verify_key` just because the `summary` is empty. Even if you could argue that it's unnecessary for those not making a game, it's still some form of intentionally designed data loss which I don't want.

I suggest reverting the second class entirely and bringing the PR back to its mono-class state.